accio-base/src/test/java/io/accio/base/sqlrewrite/analyzer/TestStatementAnalyzer.java

```diff
@@ -62,39 +86,131 @@ public void testGetTableWithoutWithTable()
+                .setSchema("test")
+                .setModels(ImmutableList.of(
+                        model("table_1", "SELECT * FROM foo", ImmutableList.of(varcharColumn("c1"), column("c2", INTEGER, null, true))),
+                        model("table_2", "SELECT * FROM bar", ImmutableList.of(varcharColumn("c1"), column("c2", DATE, null, true)))))
+                .build();
+
+        CatalogSchemaTableName t1 = new CatalogSchemaTableName("test", "test", "table_1");
+        CatalogSchemaTableName t2 = new CatalogSchemaTableName("test", "test", "table_2");
+        assertThat(analyzeSql("SELECT t1.c1, t2.c1, t2.c2 FROM table_1 t1 JOIN table_2 t2 ON t1.c2 = t2.c1\n" +
+                "WHERE t1.c1 = 'foo' AND t1.c2 >= 123 OR t2.c1 != 'bar' OR t2.c2 < DATE '2020-01-01' AND t1.c1 != t2.c1", manifest).getSimplePredicates())
+                .containsExactlyInAnyOrderElementsOf(List.of(
+                        new SimplePredicate(t1, "c1", ComparisonExpression.Operator.EQUAL, new StringLiteral("foo")),
+                        new SimplePredicate(t1, "c1", ComparisonExpression.Operator.NOT_EQUAL, DereferenceExpression.from(QualifiedName.of("t2", "c1"))),
+                        new SimplePredicate(t1, "c2", ComparisonExpression.Operator.GREATER_THAN_OR_EQUAL, new LongLiteral("123")),
+                        new SimplePredicate(t2, "c1", ComparisonExpression.Operator.NOT_EQUAL, new StringLiteral("bar")),
+                        new SimplePredicate(t2, "c2", ComparisonExpression.Operator.LESS_THAN, new GenericLiteral("DATE", "2020-01-01"))));
+    }
+
+    private Analysis analyzeSql(String sql, Manifest manifest)
+    {
+        Statement statement = SQL_PARSER.createStatement(sql, new ParsingOptions(AS_DECIMAL));
+        Analysis analysis = new Analysis(statement);
+        analyze(
+                analysis,
+                statement,
+                DEFAULT_SESSION_CONTEXT,
+                fromManifest(manifest));
+        return analysis;
+    }
+
+    @Test
+    public void testScope()
+    {
+        SessionContext sessionContext = SessionContext.builder().setCatalog("test").setSchema("test").build();
+        Manifest manifest = Manifest.builder()
+                .setCatalog("test")
+                .setSchema("test")
+                .setModels(ImmutableList.of(
+                        model("table_1", "SELECT * FROM foo", ImmutableList.of(varcharColumn("c1"), varcharColumn("c2"))),
+                        model("table_2", "SELECT * FROM bar", ImmutableList.of(varcharColumn("c1"), varcharColumn("c2")))))
+                .build();
+
+        Function<String, Scope> analyzeSql = (sql) -> {
+            Statement statement = SQL_PARSER.createStatement(sql, new ParsingOptions(AS_DECIMAL));
+            Analysis analysis = new Analysis(statement);
+            return analyze(
+                    analysis,
+                    statement,
+                    sessionContext,
+                    fromManifest(manifest));
+        };
```

Comment:
My bad, I should know the different from them. I originally expected the new method can include all case. If it just include two case may be not worth extract a method.

Reply:
It's ok. Extracting them is not bad.